cassiopeia/spatial/spatial_utilities.py

```diff
@@ -0,0 +1,116 @@
+"""
+Utilities for spatial lineage-tracing module.
+"""
+
+from typing import Optional, Tuple
+
+import anndata
+import networkx as nx
+import numpy as np
+import pandas as pd
+import squidpy as sq
+
+
+def get_spatial_graph_from_anndata(
```

Comment:
What value does this function provide? It's just a wrapper around `sq.gr.spatial_neighbors` with a call to `nx.from_numpy_array`. I suggest removing which will remove the squidpy requirement. Users who want to run `impute_spatial_data` can call `sq.gr.spatial_neighbors` first outside of cassiopeia.

Reply:
A few reasons why I have this function:

* This codeblock used in a few places
* I want to have a spatial graph constructor here in case folks don't want to create their spatial graphs and just use squidpy
* Having this abstracted away will be useful in case we want to re-write it.

It might be something we get rid of in the future, but right now I don't think it hurts to have and just adds functionality for users.